### PR TITLE
[GOBBLIN-1174] Fail job on FileBasedSource ls invalid source directory

### PR DESCRIPTION
**Pull request by zxcware**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title.
    - https://issues.apache.org/jira/browse/GOBBLIN-1174

### Description
- [x] Here are some details about my PR:
  - Today, a gobblin job that uses a FileBasedSource will succeed on an invalid `ConfigurationKeys.SOURCE_FILEBASED_DATA_DIRECTORY`, for example the path doesn't exist. In this case, the job should fail to signal wrong configuration.

### Tests
- [x] My PR adds the following unit tests:
  - `FileBasedSourceTest.testFailOnInvalidSourceDirectory` verifies that a `RuntimeException` was thrown with an invalid path

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by ZihanLi58** [comment]
+1 LGTM

File: gobblin-core/src/main/java/org/apache/gobblin/source/extractor/hadoop/AvroFileSource.java [hunk 1]
```diff
@@ -28,13 +28,16 @@
...
 import org.apache.gobblin.configuration.ConfigurationKeys;
 import org.apache.gobblin.configuration.State;
 import org.apache.gobblin.configuration.WorkUnitState;
 import org.apache.gobblin.source.extractor.Extractor;
 import org.apache.gobblin.source.extractor.filebased.FileBasedSource;
 
 
+@Slf4j
 public class AvroFileSource extends FileBasedSource<Schema, GenericRecord> {
   private static final Logger LOGGER = LoggerFactory.getLogger(AvroFileSource.class);
```
Comment (yukuai518) [inline gobblin-core/src/main/java/org/apache/gobblin/source/extractor/hadoop/AvroFileSource.java:42]: Can we clean up the LOGGER here if we are using the Slf4j wrapper?

Comment (zxcware) [reply gobblin-core/src/main/java/org/apache/gobblin/source/extractor/hadoop/AvroFileSource.java]: nice catch! Done.

**Comment by yukuai518** [comment]
+1 LGTM

**Pull request closed without merging**